misc/lint/rules.go

```diff
@@ -0,0 +1,37 @@
+//go:build ruleguard
+
+package gorules
+
+import "github.com/quasilyte/go-ruleguard/dsl"
+
+// cf. https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices
+func declareEmptySlices(m dsl.Matcher) {
+	m.Match(
+		`$name := []$t{}`,
+		`$name := make([]$t, 0)`,
+	).
+		Suggest(`var $name []$t`).
+		Report(`replace '$$' with 'var $name []$t'`)
+}
```

Comment:
as for me, `var $name []$t` isn't equal `$name := make([]$t, 0)`,
because `$name := make([]$t, 0)` contains two operations:
- create a new variable 
- init this one with an empty array,

but `var $name []$t` does only creating

Reply:
so after `var $name []$t`, `$name` is nil, but after `$name := make([]$t, 0)` '$name` is `[]$t`

Reply:
You're right, in this case as per the golang spec it makes sense to use `make` due to JSON being involved. I've updated it.